ros2cli/ros2cli/node/__init__.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# TODO(mikaelarguedas) revisit this once it's specified
+HIDDEN_NODE_PREFIX = '_'
+CLI_NODE_NAME_PREFIX = HIDDEN_NODE_PREFIX + 'ros2cli'
```

Comment:
Nipick: why use `CLI_` as the prefix for this variable? I would suggest either `ROS2CLI_NODE_NAME_PREFIX` (to match the package name) or just `NODE_NAME_PREFIX` (since the symbol is defined in the namespace of `ros2cli`).

Reply:
> Imo the consolidation should not happen here in ros2cli (I assume that is why you kept the TODO comment) but go into rclpy. Similar to HIDDEN_TOPIC_PREFIX.

Ok, I can expose it from `rclpy/node.py`. I kept the comment because I was moving the same piece of code to a different place, and wasn't sure what he meant by `once it's specified` :)

Reply:
Actually, thinking some more, I'm not so sure about it. I can make the change if you still think that's the way to go, but this feature is strictly at the ROS CLI level; the node is not inherently 'hidden' in any way. It's simply an interpretation given by the CLI based on a prefix. I would think exposing a variable named `HIDDEN_NODE_PREFIX` from `rclpy` might be confusing and incorrectly imply that there's an actual 'hidden node feature' implemented in ROS.

Reply:
I would argue that the very same semanstic applies to [HIDDEN_TOPIC_PREFIX](https://github.com/ros2/rclpy/blob/c108300f4068afbf6d839132a09f4dbf613b62b0/rclpy/rclpy/topic_or_service_is_hidden.py#L15) and that is exposed by `rclpy` in order to establish a common understanding of what by convention is being considered "hidden".

Reply:
> I would argue that the very same semanstic applies to [HIDDEN_TOPIC_PREFIX](https://github.com/ros2/rclpy/blob/c108300f4068afbf6d839132a09f4dbf613b62b0/rclpy/rclpy/topic_or_service_is_hidden.py#L15) and that is exposed by `rclpy` in order to establish a common understanding of what by convention is being considered "hidden".

Yep, and I'd say HIDDEN_TOPIC_PREFIX could've been defined in `ros2cli` as well. I don't feel strongly though; made the changes and updated the PR.

Reply:
> I'd say HIDDEN_TOPIC_PREFIX could've been defined in ros2cli as well.

If the underscore prefix is supposed to be a generally agreed upon convention it isn't sufficient to define it in `ros2cli`. E.g. what if `rviz` want to show a drop down list of topics and ignore hidden ones? `ros2cli` isn't accessible for that. So it needs to be defined in the client library across all languages.

In the future the actual definition could be pushed down into a constant defined in a `.msg` / `.idl` file.